merge/merge_test.go

```diff
@@ -0,0 +1,127 @@
+package diff
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestMergeReplaced(t *testing.T) {
+	cases := []struct {
+		Case        string
+		Prev        map[string]interface{}
```

Comment:
Some of these tests might be easier to read by writing the test cases as raw JSON instead of Go types.